ahmed-yara-sara/src/components/Search.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from "react";
+import { Spinner, Form, FormControl, Button } from "react-bootstrap";
+import "bootstrap/dist/css/bootstrap.min.css";
+export default function Search(props) {
+  const [isSubmitted, submitForm] = React.useState(false);
+  const [searchValue, setSearchValue] = React.useState("");
+  const TMDB_BASE_URL = "https://api.themoviedb.org/3";
+  const constructUrl = (path, query) => {
+    return `${TMDB_BASE_URL}/${path}?api_key=${atob(
+      "ZDJmYTdhZDFlMjZhZjA4NDdkMzQ5ZDdkYmQ1ZjkzZTU="
+    )}&query=${query}`;
+  };
+
+  const handleChange = (event) => {
+    setSearchValue(event.target.value);
+  };
+  const onSubmit = (e) => {
+    e.preventDefault();
+    submitForm(true);
+    fetchMoives();
+  };
+  React.useEffect(() => {
+    fetchMoives();
+    if (isSubmitted)
+      setTimeout(() => {
+        submitForm(false);
+      }, 3000);
+  }, [isSubmitted]);
```

Comment:
the loading of the spinner should depend on the time taken from the fetch to get the data from the server
it should't depend on static time